src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/SourceControlManagement.java

```diff
@@ -132,11 +133,19 @@ public void verifyUIDirectedToInitialPage() {
   }
 
   @Then("Click push button in Actions dropdown")
-  public void clickEdit() {
+  public void clickPushAction () {
     ElementHelper.clickOnElement(Helper.locateElementByTestId("pipeline-actions-btn"));
     ElementHelper.clickOnElement(Helper.locateElementByTestId("push-pipeline"));
   }
 
+  @Then("Click pull button in Actions dropdown")
+  public void clickPullAction() {
+    ElementHelper.clickOnElement(Helper.locateElementByTestId("pipeline-actions-btn"));
+    ElementHelper.clickOnElement(Helper.locateElementByTestId("pull-pipeline"));
+    Commands.waitForLoading();
+    WaitHelper.waitForPageToLoad();
```

Comment:
why do we need to wait ?

Reply:
The page refreshes. If we locate an element before the page has refreshed, the reference becomes invalid after the re-load.